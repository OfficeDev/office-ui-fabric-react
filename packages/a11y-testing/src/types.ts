export type PropValue = string | number | boolean | any;
export type Props = { [name: string]: PropValue };
export type AccessibilityBehavior = (props: Props) => any;

export interface Rule {
  stringify: () => string;
}

export interface Slot {
  name: string;
  props: Props[];
  expectAttribute: boolean;
  expectedAttribute: string;
  expectedValue: PropValue;
  description: string;
  hidden: boolean;
  afterEvent: string;
  afterEventData: any[];
  checkEvent: boolean;
}

export interface TestFacade {
  slotExists(slotName: string): boolean;
  attributeExists(slotName: string, attributeName: string): boolean;
  attributeHasValue(slotName: string, attributeName: string, value: PropValue): boolean;
  getAttributeValue(slotName: string, attribute: string): PropValue;
  forProps(props: Props): TestFacade;
  afterEvent(slotName: string, eventName: string, args: any[]): void;
}
