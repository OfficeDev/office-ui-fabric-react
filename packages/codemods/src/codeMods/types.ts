import { SourceFile } from 'ts-morph';

// tslint:disable-next-line: interface-name
export interface CodeModResult {
  success?: boolean;
}

// tslint:disable-next-line: interface-name
export interface Codemod<T = SourceFile> {
  /**
   * The version before which the codemod should run.
   * Must be valid semver.
   * Currently has no impact. Maybe remove?
   */
  version?: string;
  /**
   * A string to help identify the codemod.
   */
  name: string;
  /**
   * The actual function that should be run on any given file.
   * TODO, is there a possibility of codemods that would need to execute over mutiple files?
   */
  run: (file: T) => CodeModResult;
  /**
   * If not enabled, then this mod will not be conisdered to run. Only enable it once it's ready
   * to be applied in real world scenarios
   */
  enabled?: boolean;
}
