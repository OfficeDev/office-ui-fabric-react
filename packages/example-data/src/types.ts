export interface IFooProps {
  /** @deprecated */
  dep1?: string;
  /** @deprecated */
  dep2?: string;
  notDep?: string;
}
