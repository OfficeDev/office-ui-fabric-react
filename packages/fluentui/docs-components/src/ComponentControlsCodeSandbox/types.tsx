export type CodeSandboxImport = {
  module: any;
  version: string;
  required: boolean;
};

export type SourceManagerLanguage = 'js' | 'ts';
