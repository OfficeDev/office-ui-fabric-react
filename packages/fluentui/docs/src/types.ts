import { UseKnobOptions } from '@fluentui/docs-components';
import { ThemePrepared } from '@fluentui/react-northstar';
import { FluentComponentInfo, ComponentProp } from '@fluentui/react-docgen';

// !!!!!!!!!!
// Keep in sync with scripts/docs/types.ts!
// !!!!!!!!!!
export type ExampleSource = {
  js: string;
  ts: string;
};

/**
 * Component menu item shown in the site sidebar.
 * An array of menu items is saved in `componentMenu.json`.
 */
// !!!!!!!!!!
// Keep in sync with scripts/docs/types.ts!
// !!!!!!!!!!
export type ComponentMenuItem = {
  displayName: string;
  type: string;
};

export type KnobGeneratorOptions = {
  propName?: string;
  propDef: ComponentProp;
  componentInfo: FluentComponentInfo;
  theme: ThemePrepared;
};
export type KnobDefinition = UseKnobOptions<any> & { hook: Function };

export type KnobGenerator<T> = (options: KnobGeneratorOptions) => KnobDefinition;

export type KnobComponentGenerators<P> = Partial<Record<keyof P, KnobGenerator<any>>>;
