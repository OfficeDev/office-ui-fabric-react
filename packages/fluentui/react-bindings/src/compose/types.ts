import * as React from 'react';

import { COMPOSE_PROP } from './compose';

export type ComposeInputComponent<P = {}> = React.FunctionComponent<P> & { [COMPOSE_PROP]?: ComposePreparedOptions };

export type ComposeInputOptions<
  AllProps extends Record<string, any>,
  BehaviorProps extends Record<string, any>,
  StylesProps extends Record<string, any>
> = {
  className?: string;
  displayName?: string;

  mapPropsToBehavior?: (props: AllProps) => BehaviorProps;
  mapPropsToStyles?: (props: AllProps) => StylesProps;

  handledProps?: (keyof AllProps)[];
  overrideStyles?: boolean;
};

export type ComposePreparedComponent<P = {}> = React.FunctionComponent<P> & { [COMPOSE_PROP]: ComposePreparedOptions };

export type ComposePreparedOptions = {
  className: string;
  displayNames: string[];

  mapPropsToBehaviorChain: ((props: Record<string, any>) => Record<string, any>)[];
  mapPropsToStylesChain: ((props: Record<string, any>) => Record<string, any>)[];

  handledProps: (keyof Record<string, any>)[];
  overrideStyles: boolean;
};

export type ComposeResolvedOptions = {
  className: string;
  displayNames: string[];

  behaviorProps: Record<string, any>;
  stylesProps: Record<string, any>;

  handledProps: (keyof Record<string, any>)[] | undefined;
  overrideStyles: boolean;
};
