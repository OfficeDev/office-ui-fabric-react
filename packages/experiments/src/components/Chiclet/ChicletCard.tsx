import * as React from 'react';
import {
  BaseComponent
} from '../../Utilities';
import { IChicletCardProps } from './Chiclet.types';

export class ChicletCard extends BaseComponent<IChicletCardProps, any> {
  public render() {
    const { title, description, image, imageType, url } = this.props;

    return (
      <div> // style these
        { title ? (<div>{ title }</div>) : (null) }
        { image ? (<div>{ image }</div>) : (null) }
        { imageType ? (<div>{ imageType }</div>) : (null) }
        { url ? (<div>{ url }</div>) : (null) }
        { description ? (<div>{ description }</div>) : (null) }
        ...
    </div >
    );
  }

}