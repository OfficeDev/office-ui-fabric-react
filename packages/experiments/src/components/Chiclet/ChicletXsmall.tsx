import { styled } from '../../Utilities';
import { IChicletXsmallProps, IChicletXsmallStyleProps, IChicletXsmallStyles } from './ChicletXsmall.types';
import { getStyles } from './ChicletXsmall.styles';
import { ChicletXsmallBase } from './ChicletXsmall.base';

export const ChicletXsmall: React.StatelessComponent<IChicletXsmallProps> = styled<
  IChicletXsmallProps,
  IChicletXsmallStyleProps,
  IChicletXsmallStyles
>(ChicletXsmallBase, getStyles, undefined, {
  scope: 'ChicletXsmall'
});
