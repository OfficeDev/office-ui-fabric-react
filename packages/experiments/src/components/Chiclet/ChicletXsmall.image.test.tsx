import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { ChicletXsmall } from './ChicletXsmall';
import { IChicletCardProps } from './ChicletCard.types';

describe('Dialog', () => {
  it('renders chiclet with title, icon, onClick, and url', () => {
    const chicletCardProps: IChicletCardProps = {
      title: 'My Daily Notes',
      itemType: 'docx',
      image: 'https://imaging.nikon.com/lineup/dslr/df/img/sample/img_01.jpg',
      onClick: () => alert('test')
    };
    const component = renderer.create(<ChicletXsmall {...chicletCardProps} />);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
