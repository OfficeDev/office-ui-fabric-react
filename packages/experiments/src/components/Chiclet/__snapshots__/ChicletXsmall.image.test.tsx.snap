// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog renders chiclet with title, icon, onClick, and url 1`] = `
<div
  className=
      ms-ChicletXsmall
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background-color: #ffffff;
        box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.3);
        box-sizing: border-box;
        display: flex;
        flex-wrap: wrap;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 10px;
        font-weight: 400;
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        overflow: hidden;
        padding-bottom: 4px;
        padding-left: 4px;
        padding-right: 4px;
        padding-top: 4px;
        position: relative;
        user-select: none;
        width: 250px;
      }
      &:hover {
        cursor: pointer;
      }
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className=
        ms-ChicletXsmall-preview
        {
          flex-basis: 25%;
        }
  >
    <div
      className=
          ms-Image
          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: 14px;
            font-weight: 400;
            overflow: hidden;
          }
      style={
        Object {
          "height": undefined,
          "width": undefined,
        }
      }
    >
      <img
        className=
            ms-Image-image
            ms-Image-image--portrait
            is-notLoaded
            is-fadeIn
            {
              display: block;
              opacity: 0;
            }
        onError={[Function]}
        onLoad={[Function]}
        role="presentation"
        src="https://imaging.nikon.com/lineup/dslr/df/img/sample/img_01.jpg"
      />
    </div>
  </div>
  <div
    className=
        ms-ChicletXsmall-titleBox
        {
          display: flex;
          flex-basis: 70%;
          flex-wrap: wrap;
          margin-left: 4px;
        }
  >
    <div
      className=
          ms-ChicletXsmall-title
          {
            color: #323130;
            font-size: 14px;
            height: 37px;
            letter-spacing: normal;
            line-height: 1.25;
            overflow: hidden;
            text-align: left;
            width: 170px;
            word-wrap: break-word;
          }
    >
      My Daily Notes
    </div>
    <div
      className=

          {
            font-size: 12px;
            height: 17px;
            overflow: hidden;
            text-overflow: ellipsis;
            white-space: nowrap;
            width: 170px;
          }
    />
  </div>
</div>
`;
